zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -205,6 +220,28 @@ public Repository(String keyspace, Cluster cluster, Boolean ensureSchema) {
                     .value("ts", QueryBuilder.bindMarker("ts"))
                     .value("trace_id", QueryBuilder.bindMarker("trace_id"))
                     .using(QueryBuilder.ttl(QueryBuilder.bindMarker("ttl_"))));
+
+        selectTraceIdsBySpanDuration = session.prepare(
+                QueryBuilder.select("d", "ts", "tid")
+                    .from("span_duration_index")
+                    .where(QueryBuilder.eq("s", QueryBuilder.bindMarker("service_name")))
```

Comment:
I'm aware of the problem and that it affects both heap and sstable size.

In the heap it's not so much about space but objection creation/deletion, IIRC.
And in the sstable it is kinda crazy that every column value is labeled with its column name in every single row through the file. (It's a hangup from when Cassandra used thrift and was truly schema-free).

And I've been aware that it's getting addressed between CASSANDRA-8099 and CASSANDRA-7447

But those numbers above look surprising for what you say is off a simple schema.
Just how long were the long column names? And was compression enabled (which usually makes up for the difference)?

Otherwise it's evidence on sstable size, not performance.
And if we don't have a problem with that (we want to encourage people to scale out rather than scale up) it can well be a better optimisation of developer time to use meaningful column names.

TL;DR i'd love to hear more around the details of from what your metrics team discovered.